DOCS-10243 (#28651)
DataDog/documentation: content/en/tracing/trace_collection/library_config/_index.md

**Supported Input**: Boolean <br>
**Description**: Enables or disables sending traces from the application.

`DD_TRACE_SAMPLE_RATE`
: **Default**: `-1`. If unset, the tracer defers to the Datadog Agent to control sample rate. <br>
**Supported Input**: A float between 0.0 and 1.0, inclusive.  <br>
**Caveats**: This variable is deprecated in favor of `DD_TRACE_SAMPLING_RULES`, which provides more flexible and granular sampling control.  <br>

> **Description**: Controls the trace ingestion sample rate between the Datadog Agent and the backend. Must be a float between 0.0 and 1.0, where 1.0 means all traces are sent to the backend and 0.0 means none are sent. This setting applies globally to all traces and does not support per-service or per-operation targeting.

Reviewer: ```suggestion
**Description**: Controls the trace ingestion sample rate between the Datadog Agent and the backend. Must be a number between 0.0 and 1.0, where 1.0 means all traces are sent to the backend and 0.0 means none are sent. This is precise up to 6 digits, applies globally to all traces, does not support per-service or per-operation targeting. 
```

I also added a bit about the precision here since anything past 6 digits can run into floating point rounding issues where the sampling rate may be slightly over/under, but not sure if the customer would care

> **Description**:  Controls the trace ingestion sample rate between the Datadog Agent and the backend. Must be a float between 0.0 and 1.0, where 1.0 means all traces are sent to the backend and 0.0 means none are sent. This setting applies globally to all traces and does not support per-service or per-operation targeting.

Reviewer 2: ```suggestion
**Description**: Controls the trace ingestion sample rate between the Datadog Agent and the backend. Must be a float between 0.0 and 1.0, where 1.0 means all traces are sent to the backend and 0.0 means none are sent. This setting applies globally to all traces and does not support per-service or per-operation targeting.
```
Just making Vale happy

`DD_TRACE_SAMPLING_RULES`
: **Default**: `null`. If unset or no rules match, the tracer defers to the Datadog Agent to dynamically adjust sample rate across traces.  <br>

> **Supported Input**: A JSON array of objects. <br>

Reviewer: Wondering if we should change the wording here to `A JSON array of SamplingRule.` and define that below along with the types of `service`, `name`, `resource`, `tags`, and `max_per_second`. Not sure if that would fit on this page though

Maybe this should just link to the sampling page with the specifics

@dussault-antoine wdyt?

Reviewer 3: Yep, I would suggest to simply link to the ingestion control page: https://docs.datadoghq.com/tracing/trace_pipeline/ingestion_mechanisms/?tab=java#in-tracing-libraries-user-defined-rules

Reviewer: I'd suggest a different link on line 51, something like the below (if possible)

```suggestion
**Supported Input**: A JSON array of [user-defined rules](https://docs.datadoghq.com/tracing/trace_pipeline/ingestion_mechanisms/?tab=java#in-tracing-libraries-user-defined-rules). <br>
```

> **Caveats**: Matching by `resource` and `tags` is in Preview.  <br>

Reviewer 3: This is GA indeed 👍

**Description**: Enables fine-grained control over trace ingestion, allowing you to target specific services, operations, resources, or tagged traces. Defined by a JSON array of objects, where each object must include a `sample_rate` between 0.0 and 1.0 (inclusive), and can optionally include fields such as `service`, `name`, `resource`, `tags`, and `max_per_second`. Objects are evaluated in the order listed; the first matching object determines the trace's sample rate. For more information, see [Ingestion Mechanisms][4]. <br>

> **Examples**: <br>

Reviewer: ```suggestion
**Use cases**: <br>
```

@duncanhewett wdyt about this verbiage?

Reviewer 4: I think examples is fine personally, but happy either way!

[1]: /developers/community/libraries/#apm-tracing-client-libraries
[2]: /tracing/trace_collection/compatibility/java/#framework-integrations-disabled-by-default
[3]: /tracing/services/inferred_services/

> [4]: /tracing/trace_pipeline/ingestion_mechanisms/?tab=net#pagetitle

Reviewer 2: ```suggestion
[4]: /tracing/trace_pipeline/ingestion_mechanisms/?tab=net#pagetitle
```
The `#pagetitle` hash seems odd, did you mean to link to something else?
